packages/rum-core/src/boot/rumPublicApi.ts

```diff
@@ -201,40 +217,45 @@ export function makeRumPublicApi(
     }),
 
     setUser: monitor((newUser: User) => {
-      const sanitizedUser = sanitizeUser(newUser)
-      if (sanitizedUser) {
-        user = sanitizedUser
-      } else {
+      if (typeof newUser !== 'object' || !newUser) {
         display.error('Unsupported user:', newUser)
+      } else {
+        userContextManager.setContext(sanitizeUser(newUser as Context))
       }
     }),
 
-    removeUser: monitor(() => {
-      user = {}
+    getUser: monitor(userContextManager.getContext),
+
+    setUserProperty: monitor((key, property) => {
+      const sanitizedProperty = sanitizeUser({ [key]: property })[key]
+      userContextManager.setContextProperty(key, sanitizedProperty)
```

Comment:
<!-- Nitpicks are small, trivial, but necessary changes. Distinguishing nitpick comments significantly helps direct the reader's attention to comments requiring more involvement.  -->
**🥜 nitpick:** ‏ ‏This is a bit weird. We could have a `sanitizeUserProperty(key, value)` function that cast the value if key is `id`, `name` or `email` instead, and use it in `sanitizeUser`. But it's not dramatic and we can live with it :)

Reply:
I agree this isn't the nicest, but I really wanted to avoid having 2x the same sanitize function for different input types. Having once source of truth is also nicer if we decide to expand `sanitizeUser`